presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonToArrayCast.java

```diff
@@ -86,7 +87,7 @@ public static Block toArray(Type arrayType, ConnectorSession connectorSession, S
             return blockBuilder.build();
         }
         catch (RuntimeException e) {
-            throw new PrestoException(INVALID_CAST_ARGUMENT, "Value cannot be cast to " + arrayType, e);
+            throw new PrestoException(INVALID_CAST_ARGUMENT, format("'%s' cannot be cast to %s", json.toStringUtf8(), arrayType), e);
```

Comment:
Move the exception message from jsonParse to here as we are skipping that function now